devito/types/grid.py

```diff
@@ -22,6 +24,8 @@
 
 GlobalLocal = namedtuple('GlobalLocal', 'glb loc')
 
+_subdomain_count = {}  # Track attachment of grids to SubDomains for dimension labelling
```

Comment:
Not sure if this is good to have! Need to check!

Reply:
it isn't

Reply:
we have to fix it